src/trigger.py

```diff
@@ -82,25 +45,78 @@ class cmd_run(Command):
     def __init__(self, sub_parser, name):
         super().__init__(sub_parser, name)
         self._logger = Logger("config/logger.conf", "trigger")
+        self._build_configs = None
+        self._db = None
+
+    def _log_revision(self, message, build_config, head_commit):
+        self._logger.log_message(
+            logging.INFO,
+            f"{message:32s} {build_config.name:32s} {head_commit}"
+        )
+
+    def _run_trigger(self, build_config, force):
+        head_commit = kernelci.build.get_branch_head(build_config)
+        node_list = self._db.get_nodes({
+            "revision.commit": head_commit,
+        })
+
+        if node_list:
+            if force:
+                self._log_revision(
+                    "Resubmitting existing revision", build_config, head_commit
+                )
+            else:
+                self._log_revision(
+                    "Existing revision", build_config, head_commit
+                )
+                return
+        else:
+            self._log_revision(
+                "New revision", build_config, head_commit
+            )
+
+        revision = {
+            'tree': build_config.tree.name,
+            'url': build_config.tree.url,
+            'branch': build_config.branch,
+            'commit': head_commit,
+        }
+        node = {
+            'name': 'checkout',
+            'path': ['checkout'],
+            'revision': revision,
+        }
+        self._db.submit({'node': node})[0]
```

Comment:
No need to fetch anything further from the response as it is not being returned.
```suggestion
        self._db.submit({'node': node})
```

Reply:
I will add this change in a follow-up PR.